core/src/main/scala/ox/channels/SourceOps.scala

```
@@ -565,6 +565,61 @@ trait SourceOps[+T] { this: Source[T] =>
         case ChannelClosed.Error(r) => throw r.getOrElse(new NoSuchElementException("getting head failed"))
         case t: T @unchecked        => t
     }
+
+  /** Returns the last element from this source wrapped in `Some` or `None` when the source is empty or fails during the `receive()``
+    * operation. Note that `lastOption` is a terminal operation leaving the source in `ChannelClosed.Done` state.
+    *
+    * @return
+    *   A `Some(last element)` if source is not empty or None` otherwise.
+    * @example
+    *   {{{
+    *   import ox.*
+    *   import ox.channels.Source
+    *
+    *   supervised {
+    *     Source.empty[Int].lastOption()  // None
+    *     val s = Source.fromValues(1, 2)
+    *     s.lastOption()                  // Some(2)
+    *     s.receive()                     // ChannelClosed.Done
+    *   }
+    *   }}}
+    */
+  def lastOption(): Option[T] = Try(last()).toOption
```

shouldn't `lastOption` throw an exception, if there's an error, and only return `None` on empty (done) sources? That way we suppress the exception, and that's never a good thing

also, then we can implement `last` in terms of `lastOption`, not the other way round, and avoid throwing-catching

---

done